py/desispec/redshiftqa/desi_per_fiber_qa_plots.py

```diff
@@ -0,0 +1,257 @@
+
+from __future__ import division, print_function
+import sys, os, glob, time, warnings, gc
+import numpy as np
+import matplotlib.pyplot as plt
+from astropy.table import Table, vstack, hstack, join
+import fitsio
+# from astropy.io import fits
+
+from datetime import datetime, timedelta
+from desitarget.targetmask import desi_mask, bgs_mask
+
+import argparse
+
+import time
+time_start = time.time()
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument('-i', '--indir', type=str, help="input directory path", required=True)
+parser.add_argument('-o', '--output', type=str, help="output directory path", required=True)
+parser.add_argument('--tracer', type=str, help="tracer type, e.g., LRG", required=True)
+parser.add_argument('--stats', type=str, help="per-fiber stats path", required=True)
+args = parser.parse_args()
+
+indir = args.indir
+output_dir = args.output
+tracer = args.tracer
+stats_fn = args.stats
+
+pvalue_threshold = 1e-4
+save_pdf = False
+save_png = True
+
+stats = Table(fitsio.read(stats_fn))
+
+print(tracer)
+
+if tracer in ['LRG', 'ELG', 'QSO', 'ELG_LOP', 'ELG_VLO', 'BGS_ANY']:
+    fn = os.path.join(indir, 'ztile-main-dark-cumulative.fits')
+    fn1 = os.path.join(indir, 'ztile-main-dark-cumulative-extra.fits')
+    cat = Table(fitsio.read(fn))
+    cat1 = Table(fitsio.read(fn1, columns=['Z', 'ZWARN', 'Z_QSO', 'GOOD_Z_LRG', 'GOOD_Z_ELG', 'GOOD_Z_QSO']))
+    cat = hstack([cat, cat1], join_type='exact')
+    mask = np.where(cat['DESI_TARGET'] & desi_mask[tracer] > 0)[0]
+    cat = cat[mask]
+else:
+    fn = os.path.join(indir, 'ztile-main-bright-cumulative.fits')
+    fn1 = os.path.join(indir, 'ztile-main-bright-cumulative-extra.fits')
+    cat = Table(fitsio.read(fn))
+    cat1 = Table(fitsio.read(fn1, columns=['GOOD_Z_BGS']))
+    cat = hstack([cat, cat1], join_type='exact')
+    mask = np.where(cat['BGS_TARGET'] & bgs_mask[tracer] > 0)[0]
+    cat = cat[mask]
+
+print(tracer, 'zcat', len(cat))
+stats_zcat = Table()
+stats_zcat['FIBER'], stats_zcat[tracer.lower()+'_zcat_n_tot'] = np.unique(cat['FIBER'], return_counts=True)
+
+# Remove FIBERSTATUS!=0 fibers
+mask = cat['COADD_FIBERSTATUS']==0
+print('FIBERSTATUS   ', np.sum(~mask), np.sum(mask), np.sum(~mask)/len(mask))
+cat = cat[mask]
+
+# Remove "no data" fibers
+mask = cat['ZWARN'] & 2**9==0
+print('No data   ', np.sum(~mask), np.sum(mask), np.sum(~mask)/len(mask))
+cat = cat[mask]
+
+# Require a minimum depth
+if tracer in ['BGS_ANY', 'BGS_BRIGHT', 'BGS_FAINT']:
+    min_depth = 160
+    mask = cat['EFFTIME_SPEC']>min_depth
+else:
+    min_depth = 800.
+    mask = cat['EFFTIME_SPEC']>min_depth
+print('Min depth   ', np.sum(~mask), np.sum(mask), np.sum(~mask)/len(mask))
+cat = cat[mask]
+
+# Apply masks
+if tracer=='LRG':
+    lrgmask = Table(fitsio.read('/dvs_ro/cfs/cdirs/desi/users/rongpu/targets/dr9.0/1.1.1/resolve/dr9_lrg_1.1.1_lrgmask_v1.1_with_targetid.fits'))
```

Comment:
We shouldn't hardcode NERSC-specific or user-specific paths.  Is there a generic location under $DESI_ROOT that could be used instead?  Same for elgmask and other references to "cfs" and "rongpu" in other scripts.

Reply:
The scripts now point to the equivalent files in the LSS directory.